### DISCUSSION TITLE
webhook send_message_response

### DISCUSSION THREAD

**Discussion by azilin** [kirimwa-id/forum#89] [Tanya Jawab]
halo bang rio,
saya membuat halaman webhook agar dpt notif setelah pengiriman pesan, kemudian akan disimpan di table log, saya mencoba beberapa kali, belum berhasil masuk ke table 
mohon saran bang .. terimakasih 
       
      `$postdata = @file_get_contents("php://input");
        $event = json_decode($postdata);
        if ($event->webhook_type == 'send_message_response') {
         
         // simpan respon ke table log     
            $model = new ModelApp();
            $data = array(
                'response_id' => $event->id,
                'webhook_type' => $event->webhook_type,
                'status' => $event->status,
                'message' => $event->message
            );
            $model->save('tb_msg_log',$data);  
        }`

**Comment by rioastamal** [comment]
Halo @azilin

1. Pastikan sudah menambahkan webhook url lewat [POST /v1/webhooks](https://developer.kirimwa.id/api/v1/#post-webhooks)
2. Cek response dari webhook API KirimWA.id, apakah memang benar terkirim ke alamat webhook url yang sudah ditentukan menggunakan API [GET /v1/webhook-responses](https://developer.kirimwa.id/api/v1/#get-webhook-responses).

Untuk melihat semua webhook `send_message_response` yang sukses dikirim gunakan.

```sh
$ curl 'https://api.kirimwa.id/v1/webhook-responses?webhook_type=send_message_response&status=success' \
  -H 'Authorization: Bearer API_TOKEN'
```

Untuk melihat semua webhook `send_message_response` yang gagal dikirim gunakan.

```sh
$ curl 'https://api.kirimwa.id/v1/webhook-responses?webhook_type=send_message_response&status=fail' \
  -H 'Authorization: Bearer API_TOKEN'
```

**Reply by azilin** [reply]
iya bang, kurang teliti saya, ternyata ada typo di webhook url 
sudah bisa, terimakasih banyak
